### PR TITLE
8339696: Clarify modeling scope of javax.lang.model.element

### PR DESCRIPTION
**Pull request by jddarcy**
Clarify that while javax.lang.model does not include an AST API, it is intended to be usable by an AST API.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339696](https://bugs.openjdk.org/browse/JDK-8339696): Clarify modeling scope of javax.lang.model.element (**Enhancement** - P4)

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) Review applies to [2c213737](https://git.openjdk.org/jdk/pull/20900/files/2c213737c92043e5eab9786404cb543c93ba1929)
 * @abdelhak-zaaim (no known openjdk.org user name / role) Review applies to [2c213737](https://git.openjdk.org/jdk/pull/20900/files/2c213737c92043e5eab9786404cb543c93ba1929)
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**) Review applies to [2c213737](https://git.openjdk.org/jdk/pull/20900/files/2c213737c92043e5eab9786404cb543c93ba1929)
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20900/head:pull/20900` \
`$ git checkout pull/20900`

Update a local copy of the PR: \
`$ git checkout pull/20900` \
`$ git pull https://git.openjdk.org/jdk.git pull/20900/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20900`

View PR using the GUI difftool: \
`$ git pr show -t 20900`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20900.diff">https://git.openjdk.org/jdk/pull/20900.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20900#issuecomment-2334903030)

### CODE REVIEW

**Review by lahodaj — APPROVED** [review]
Looks good to me.

**Review by abdelhak-zaaim — APPROVED** [review]

**Review by pavelrappo — APPROVED** [review]
Looks good.

File: src/java.compiler/share/classes/javax/lang/model/element/package-info.java [hunk 1]
```diff
@@ -32,9 +32,15 @@
  * The interfaces in this package do not model the structure of a
  * program inside a method body; for example there is no
  * representation of a {@code for} loop or {@code try}-{@code finally}
- * block.  However, the interfaces can model some structures only
- * appearing inside method bodies, such as local variables and
- * anonymous classes.
+ * block. Concretely, there is no model of the abstract syntax tree
```
Comment (jonathan-gibbons) [inline src/java.compiler/share/classes/javax/lang/model/element/package-info.java:35]: I suggest changing `the` (definite article) to something less definite, like `any`

**Review by jonathan-gibbons — APPROVED** [review]

**Pull request closed without merging**